appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java

```diff
@@ -1017,7 +1019,10 @@ public void resourceClosed(ResourceHandle h)
         }
 
         if (!state.isBusy()) {
-            throw new IllegalStateException("state.isBusy() : false");
+            //throw new IllegalStateException("state.isBusy() : false");
+            MultiException noBusyException = new MultiException(state.getBusyStackException());
```

Comment:
And then you will need handle an option of busy stack exception being null. The exception message may suggest to set the system property.